Dockerfile

```diff
@@ -18,34 +18,32 @@ ARG BASE=golang:1.16-alpine3.12
 FROM ${BASE} AS builder
 
 ARG MAKE='make build'
-ARG ALPINE_PKG_BASE="make git"
+ARG ALPINE_PKG_BASE="make git openssh-client gcc libc-dev zeromq-dev libsodium-dev"
 ARG ALPINE_PKG_EXTRA=""
 
 RUN sed -e 's/dl-cdn[.]alpinelinux.org/nl.alpinelinux.org/g' -i~ /etc/apk/repositories
 RUN apk add --update --no-cache ${ALPINE_PKG_BASE} ${ALPINE_PKG_EXTRA}
 
-WORKDIR $GOPATH/src/github.com/edgexfoundry/device-snmp-go
+WORKDIR /device-snmp-go
 
-COPY go.mod .
-COPY Makefile .
+COPY . .
 
+RUN go mod tidy
 RUN make update
 
-COPY . .
 RUN ${MAKE}
 
 FROM alpine:3.12
 
-# Make sure you change the cmd/res/configuration.toml port if you change it here
-ENV APP_PORT=49993
-EXPOSE $APP_PORT
+# dumb-init needed for injected secure bootstrapping entrypoint script when run in secure mode.
+RUN apk add --update --no-cache zeromq dumb-init
 
-ENV DEVICE_PORT=161
-EXPOSE $DEVICE_PORT
+COPY --from=builder /device-snmp-go/cmd /
+COPY --from=builder /device-snmp-go/Attribution.txt /
+COPY --from=builder /device-snmp-go/LICENSE /
 
-COPY --from=builder /go/src/github.com/edgexfoundry/device-snmp-go/cmd /
-COPY --from=builder /go/src/github.com/edgexfoundry/device-snmp-go/Attribution.txt /
-COPY --from=builder /go/src/github.com/edgexfoundry/device-snmp-go/LICENSE /
+EXPOSE 49993
+EXPOSE 161
 
 ENTRYPOINT ["/device-snmp-go"]
```

Comment:
Please update the makefile to produce `device-snmp` instead of `device-snmp-go` and change exactable name it here.

As part of the service key names we have also be updating the executables to be consistent names.